streams/src/test/java/org/apache/kafka/streams/kstream/internals/KCogroupedStreamImplTest.java

```diff
@@ -0,0 +1,403 @@
+            testInputTopic.pipeInput("1", "A", 0);
+            testInputTopic.pipeInput("11", "A", 0);
+            testInputTopic.pipeInput("11", "A", 0);
+            testInputTopic.pipeInput("1", "A", 0);
+        }
+
+        assertThat(processorSupplier.theCapturedProcessor().processed, equalTo(Arrays.asList(
+            new KeyValueTimestamp("1", "A", 0),
+            new KeyValueTimestamp("11", "A", 0),
+            new KeyValueTimestamp("11", "AA", 0),
+            new KeyValueTimestamp("1", "AA", 0)
+        )));
+    }
+
+
+    @Test
+    public void testCogroupEachTopicUnique() {
+
+        final KStream test = builder.stream("one", stringConsumed);
+        final KStream test2 = builder.stream("two", stringConsumed);
+
+        final KGroupedStream groupedOne = test.groupByKey();
+        final KGroupedStream groupedTwo = test2.groupByKey();
+
+        final KTable customers = groupedOne.cogroup(STRING_AGGREGATOR, Materialized.as("store"))
+            .cogroup(groupedTwo, STRING_AGGREGATOR)
+            .aggregate(STRING_INITIALIZER, Materialized.as("store1"));
+
+        customers.toStream().to("to-one", Produced.with(Serdes.String(), Serdes.String()));
+
+        builder.stream("to-one", stringConsumed).process(processorSupplier);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+
+            final TestInputTopic<String, String> testInputTopic = driver.createInputTopic("one", new StringSerializer(), new StringSerializer());
+            final TestInputTopic<String, String> testInputTopic2 = driver.createInputTopic("two", new StringSerializer(), new StringSerializer());
+            testInputTopic.pipeInput("1", "A", 0);
+            testInputTopic.pipeInput("1", "A", 1);
+            testInputTopic.pipeInput("1", "A", 10);
+            testInputTopic.pipeInput("1", "A", 100);
+            testInputTopic2.pipeInput("2", "B", 100L);
+            testInputTopic2.pipeInput("2", "B", 200L);
+            testInputTopic2.pipeInput("2", "B", 1L);
+            testInputTopic2.pipeInput("2", "B", 500L);
+            testInputTopic2.pipeInput("2", "B", 500L);
+            testInputTopic2.pipeInput("2", "B", 100L);
+        }
+
+        assertThat(processorSupplier.theCapturedProcessor().processed, equalTo(Arrays.asList(
+            new KeyValueTimestamp("1", "A", 0),
+            new KeyValueTimestamp("1", "AA", 1),
+            new KeyValueTimestamp("1", "AAA", 10),
+            new KeyValueTimestamp("1", "AAAA", 100),
+            new KeyValueTimestamp("2", "B", 100),
+            new KeyValueTimestamp("2", "BB", 200),
+            new KeyValueTimestamp("2", "BBB", 200),
+            new KeyValueTimestamp("2", "BBBB", 500),
+            new KeyValueTimestamp("2", "BBBBB", 500),
+            new KeyValueTimestamp("2", "BBBBBB", 500)
+        )));
+    }
+
+
+    @Test
+    public void testCogroupKeyMixedInTopics() {
```

Comment:
What is the purpose of this test? Seems to be the same as above?

Reply:
the keys are mixed across topics where in the other test the key sets are disjoint

Reply:
I see -- do we need both tests? Seem that this test super seed the other one?